pkg/reconciler/configuration/configuration.go

```diff
@@ -258,7 +264,9 @@ func (c *Reconciler) latestCreatedRevision(config *v1.Configuration) (*v1.Revisi
 		return list[0], nil
 	}
 
-	return nil, errors.NewNotFound(v1.Resource("revisions"), "revision for "+config.Name)
+	// Fallback to doing a get to name directly, just in case.
+	rev := resources.MakeRevision(ctx, config, c.clock)
+	return lister.Get(rev.Name)
```

Comment:
I say we only rely on the generation label and don't perform a fallback lookup with the name

Some user could have their own byo-rev naming scheme that could mess with this logic

Reply:
I see where you're coming from. Changing!